SerialAPI.h

```diff
@@ -102,9 +102,16 @@ class SerialAPI: public FB::JSAPIAuto
 
     void close()
     {
-        io.post(boost::bind(&SerialAPI::do_close, this, boost::system::error_code()));
-        m_thread.join();
-        io.reset();
+        if(m_thread.get_id()==boost::thread::id())

```

Comment:
This is the postcondition for boost::m_thread's default constructor (its id equals the default constructed id, which is an invalid value).   It will be false if m_thread has been initialized in set_option with io_service::run
It seems more intuitive to check for io.stopped() but this wasn't working for me, I think it is only set after run() is called, and then stopped.